File: java/service/iothub-service-sdk/src/main/java/com/microsoft/azure/iot/service/exceptions/IotHubException.java
/*
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project root for full license information.
 */

package com.microsoft.azure.iot.service.exceptions;

/**
 * Super class for IotHub exceptions
 */
public class IotHubException extends Exception
{
    public IotHubException() { super(); }
    public IotHubException(String message) { super(message); }
}

File: cpp/device/iothub_client/src/iothub_exception.cpp
#include <exception>
#include <string>
#include <sstream>

#include "iothub_exception.hpp"

namespace azure { namespace iot {

    exception::exception(const std::string& _exc, const std::string& _cls, const std::string& _func, const std::string &_aux):
        std::runtime_error(_cls + "." + _func + ", " + _aux + "\n")
    {}

}}
